src/jv_type_private.h

```diff
@@ -0,0 +1,89 @@
+#ifndef JV_TYPE_PRIVATE
+#define JV_TYPE_PRIVATE
+
+#include "jv_alloc.h"
+#include "jv.h"
+#include "jv_unicode.h"
+#include "util.h"
+
+#define KIND_MASK   0xF
+#define PFLAGS_MASK 0xF0
+#define PTYPE_MASK  0x70
+
+typedef enum {
+  JVP_PAYLOAD_NONE = 0,
+  JVP_PAYLOAD_ALLOCATED = 0x80,
+} payload_flags;
+
+
+#define JVP_MAKE_PFLAGS(ptype, allocated) ((((ptype) << 4) & PTYPE_MASK) | ((allocated) ? JVP_PAYLOAD_ALLOCATED : 0))
+#define JVP_MAKE_FLAGS(kind, pflags) ((kind & KIND_MASK) | (pflags & PFLAGS_MASK))
+
+#define JVP_FLAGS(j)  ((j).kind_flags)
+#define JVP_KIND(j)   (JVP_FLAGS(j) & KIND_MASK)
+
+#define JVP_HAS_FLAGS(j, flags) (JVP_FLAGS(j) == flags)
```

Comment:
Let's put parens around `flags` and `kind` here.

Also, I'd like the name to be a bit clearer. `JVP_HAS_FLAGS` would imply to me that I can ask about any particular bits on the flags, but this is actually saying "does it have only and exactly these specific flags?". Can we find a better name?

Reply:
Maybe. Or maybe not. 

This way it's HAS_FLAGS and HAS_KIND. But i'm not too married to this.

Reply:
Agreed about parens.

Naming - i'm open to suggestions